File: sdkwork-ai-sdk-typescript/src/types/music-generation-request.ts
/** Music generation request */
export interface MusicGenerationRequest {
  /** 模型ID */
  model: string;
  /** 提示词 */
  prompt: string;
  /** 时长(秒) */
  duration?: number;
  /** 格式: mp3, wav, flac */
  format?: string;
  /** 生成数量 */
  n?: number;
  /** 风格 */
  style?: string;
  /** 元数据 */
  metadata?: Record<string, string>;
  /** 负面提示词 */
  negative_prompt?: string;
  /** 响应格式: url, b64_json */
  response_format?: string;
  /** 参考音频URL */
  reference_audio?: string;
}

File: sdkwork-ai-sdk-java/src/main/java/com/sdkwork/ai/model/MusicGenerationRequest.java
package com.sdkwork.ai.model;

public class MusicGenerationRequest {
    private String model;
    private String prompt;
    private Integer duration;
    private String format;
    private Integer n;
    private String style;
    private Map<String, String> metadata;
    private String negativePrompt;
    private String responseFormat;
    private String referenceAudio;

    public String getModel() {
        return this.model;
    }
    
    public void setModel(String model) {
        this.model = model;
    }

    public String getPrompt() {
        return this.prompt;
    }
    
    public void setPrompt(String prompt) {
        this.prompt = prompt;
    }

    public Integer getDuration() {
        return this.duration;
    }
    
    public void setDuration(Integer duration) {
        this.duration = duration;
    }

    public String getFormat() {
        return this.format;
    }
    
    public void setFormat(String format) {
        this.format = format;
    }

    public Integer getN() {
        return this.n;
    }
    
    public void setN(Integer n) {
        this.n = n;
    }

    public String getStyle() {
        return this.style;
    }
    
    public void setStyle(String style) {
        this.style = style;
    }

    public Map<String, String> getMetadata() {
        return this.metadata;
    }
    
    public void setMetadata(Map<String, String> metadata) {
        this.metadata = metadata;
    }

    public String getNegativePrompt() {
        return this.negativePrompt;
    }
    
    public void setNegativePrompt(String negativePrompt) {
        this.negativePrompt = negativePrompt;
    }

    public String getResponseFormat() {
        return this.responseFormat;
    }
    
    public void setResponseFormat(String responseFormat) {
        this.responseFormat = responseFormat;
    }

    public String getReferenceAudio() {
        return this.referenceAudio;
    }
    
    public void setReferenceAudio(String referenceAudio) {
        this.referenceAudio = referenceAudio;
    }
}
